### ISSUE TITLE
Running on command line without video device

### ISSUE THREAD

**Issue by santosh-b** [SkyTemple/py-desmume#3]
I'm trying to run some simulations on a remote Linux server via command line. I have no need for a live screen input (but I will be occasionally dumping the screen as images), so I want to try to run it in a "headless" mode.

This is the error I get when I try to instantiate the emulator
```
emu = DeSmuME()
Microphone successfully inited.
DeSmuME 0.9.12 git#0 x64-JIT SSE4.2+AVX2
SoftRasterizer: Running using 32 additional threads. (Multithreading enabled.)
Error trying to initialize SDL: No available video device
Traceback (most recent call last):
  File "/misc/scratch/miniconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3417, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-3-cf54966c5578>", line 1, in <module>
    emu = DeSmuME()
  File "/misc/scratch/miniconda3/lib/python3.8/site-packages/desmume/emulator.py", line 741, in __init__
    raise RuntimeError("Failed to init DeSmuME")
```

Any ideas? Thanks

**Comment by theCapypara** [comment]
Currently the emulator initializes SDL on boot, which requires something to display. 
I might be able to make it optional, but it's possible the input processing is no longer possible then since it's based on SDL. 
I might need to look into some stuff to make this actually work.

**Comment by Minabsapi** [comment]
I was interested into running the emulator in headless mode as well, and just omitting the `emu.create_sdl_window()` part and everything related to the `DeSmuME_SDL_Window` object returned from that was enough and works perfectly

**Comment by theCapypara** [comment]
Great! I remember refactoring some stuff around this, and seems either I or the team that transitioned the emulator to SDL2 fixed the issue then.

**Comment by JVP15** [comment]
I've encountered the same error while trying to create the emulator with `emu=DeSmuME()` on a remote Linux server, and I thought I'd share my experience with it. I found what @Minabsapi is sort of true. When running it on my local machine with a graphics display, I can omit the `emu.create_sdl_window()` and everything works fine. But on a remote server (e.g. Google Colab) I always encounter the `Failed to init DeSmuME` error when trying to create the emulator. My workaround for this was to use a remote desktop to get Gnome, then I could create the emulator with or without a window.
